### PR TITLE
Fix `declaration-block-no-redundant-longhand-properties` false positives for `inherit` keyword

### LINKED ISSUE #6394: Fix `declaration-block-no-redundant-longhand-properties` false positives for `inherit` keyword
### What steps are needed to reproduce the bug?

1. Run stylint for checking the below CSS

```css
.demo {
  border-top-left-radius: 20px;
  border-top-right-radius: inherit;
  border-bottom-right-radius: inherit;
  border-bottom-left-radius: inherit;
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard"
}
```

### How did you run Stylelint?

stylelint src/**/*.css

### Which version of Stylelint are you using?

^14.13.0

### What did you expect to happen?

There's no error for this case because there are longhand properties with `inherit` value.
Per [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties#omitting_properties)

> Only the individual properties values can inherit. As missing values are replaced by their initial value, it is impossible to allow inheritance of individual properties by omitting them. The keyword inherit can be applied to a property, but only as a whole, not as a keyword for one value or another. That means that the only way to make some specific value to be inherited is to use the longhand property with the keyword inherit.

So, it's impossible to convert such set of longhand properties to shorthand and preserve initial semantic.

### What actually happened?

 5:3  ✖  Expected shorthand property "border-radius"  declaration-block-no-redundant-longhand-properties

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

If a longhand property of some shorthand property is set to `inherit`, don't treat such case as a rule violation.

### PR DESCRIPTION
**Pull request by kimulaco**
> Which issue, if any, is this issue related to?

Closes #6394

> Is there anything in the PR that needs further explanation?

I think this CSS spec is often overlooked. So, I added a description and example to the docs. https://github.com/stylelint/stylelint/commit/94a21c3dd197144ac92c4803641b86c4fb6402c0

### CODE REVIEW

File: .changeset/fair-oranges-love.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fix `declaration-block-no-redundant-longhand-properties` false positives for `inherit` keyword
```
Comment (ybiquitous) [inline .changeset/fair-oranges-love.md:5]: [suggest] Thanks for adding the changelog. In addition, could you please use the standard prefix `FIXED:`? Please see also the [maintainer guide](https://github.com/stylelint/stylelint/blob/d171517376238ff2ba33f98e9e171ee13ca0cd81/docs/maintainer-guide/pull-requests.md#merging).

```suggestion
Fixed: `declaration-block-no-redundant-longhand-properties` false positives for `inherit` keyword
```

File: lib/rules/declaration-block-no-redundant-longhand-properties/index.js [hunk 2]
```diff
@@ -20,6 +20,8 @@ const meta = {
 	url: 'https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties',
 };
 
+const longhandMustValues = ['inherit'];
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-redundant-longhand-properties/index.js:23]: [trivial] These suggestions depend on my personal preference. You can ignore them.

- How about using `Set` for a bit more efficiency?
- How about changing to a constant-like name for readability?

```suggestion
const IGNORED_VALUES = new Set(['inherit']);
```

See also:

https://github.com/stylelint/stylelint/blob/d171517376238ff2ba33f98e9e171ee13ca0cd81/lib/rules/color-named/index.js#L27

**Review by ybiquitous — COMMENTED** [review]
@kimulaco Thank you for creating the pull request. Adding a note about `inherit` to the rule doc is a nice idea!

I've left a few trivial suggestions, but this change almost looks good. 👍🏼

File: lib/rules/declaration-block-no-redundant-longhand-properties/README.md [hunk 3]
```diff
@@ -145,6 +145,18 @@ a {
...
+<!-- prettier-ignore -->
+```css
+a {
+  margin-top: 1px;
+  margin-right: 2px;
+  margin-bottom: 3px;
+  margin-left: inherit;
+}
+```
+
```
Comment (jeddy3) [inline lib/rules/declaration-block-no-redundant-longhand-properties/README.md:159]: Let's remove this as our patterns don't tend to show edge cases. 

We can add something to do _extended description_ at the top instead. For example, expand the following line

> This rule will only complain if you've used the longhand equivalent of _all_ the properties that the shorthand will set.

To:

"This rule will only complain if you've used the longhand equivalent of _all_ the properties that the shorthand will set and if their values are not `inherit`."

**Review by jeddy3 — CHANGES_REQUESTED** [review]
Thank you.

I've requested a change to the docs.

File: .changeset/fair-oranges-love.md (hunk 1 above)
Comment (kimulaco) [reply .changeset/fair-oranges-love.md]: Thanks. I fixed the prefix. https://github.com/stylelint/stylelint/pull/6419/commits/117e5f63aa66a80e58ef28223fff0bace1c5b34a

File: lib/rules/declaration-block-no-redundant-longhand-properties/index.js (hunk 2 above)
Comment (kimulaco) [reply lib/rules/declaration-block-no-redundant-longhand-properties/index.js]: Fixed to use `Set`. https://github.com/stylelint/stylelint/pull/6419/commits/39f4f273aba6354ce2ef8d4245bbf55dbb2c8b0d

File: lib/rules/declaration-block-no-redundant-longhand-properties/README.md (hunk 3 above)
Comment (kimulaco) [reply lib/rules/declaration-block-no-redundant-longhand-properties/README.md]: Thanks for review! I removed a description of edge case. https://github.com/stylelint/stylelint/pull/6419/commits/616a536d977f7a6c6ce283634a942737045fce77
Then, I updated the description to your suggestion. https://github.com/stylelint/stylelint/pull/6419/commits/f442a05438dcbcb783ee6c388f34dc801783e47a

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM!

**Review by jeddy3 — APPROVED** [review]
Thanks for making the changes. LGTM.

**Pull request merged**